packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -397,6 +396,404 @@ void main() {
+      ExpansionPanelRadio(
+        headerBuilder: (BuildContext context, bool isExpanded) {
+          return Text(isExpanded ? 'D' : 'C');
+        },
+        body: const SizedBox(height: 100.0),
+        value: 1,
+      ),
+      ExpansionPanelRadio(
+        headerBuilder: (BuildContext context, bool isExpanded) {
+          return Text(isExpanded ? 'F' : 'E');
+        },
+        body: const SizedBox(height: 100.0),
+        value: 2,
+      ),
+    ];
+
+    final List<Map<String, dynamic>> callbackHistory = <Map<String, dynamic>>[];
+    Map<String, dynamic> callbackResults;
+
+    final ExpansionPanelList _expansionListRadio = ExpansionPanelList.radio(
+      expansionCallback: (int _index, bool _isExpanded) {
+        callbackHistory.add(<String, dynamic>{
+          'index': _index,
+          'isExpanded': _isExpanded,
+        });
+      },
+      children: _demoItemsRadio,
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: SingleChildScrollView(
+          child: _expansionListRadio,
+        ),
+      ),
+    );
+
+    // Initializes with all panels closed
+    expect(find.text('A'), findsOneWidget);
+    expect(find.text('B'), findsNothing);
+    expect(find.text('C'), findsOneWidget);
+    expect(find.text('D'), findsNothing);
+    expect(find.text('E'), findsOneWidget);
+    expect(find.text('F'), findsNothing);
+
+    // open one panel
+    await tester.tap(find.byType(ExpandIcon).at(1));
+    await tester.pumpAndSettle();
+
+    // callback is invoked once with appropriate arguments
+    expect(callbackHistory.length, equals(1));
+    callbackResults = callbackHistory[callbackHistory.length - 1];
+    expect(callbackResults['index'], equals(1));
+    expect(callbackResults['isExpanded'], equals(false));
+
+    // close a different panel
+    await tester.tap(find.byType(ExpandIcon).at(2));
+    await tester.pumpAndSettle();
+
+    // callback is invoked the first time with correct arguments
+    expect(callbackHistory.length, equals(3));
+    callbackResults = callbackHistory[callbackHistory.length - 2];
+    expect(callbackResults['index'], equals(2));
+    expect(callbackResults['isExpanded'], equals(false));
+
+    // callback is invoked the second time with correct arguments
+    callbackResults = callbackHistory[callbackHistory.length - 1];
+    expect(callbackResults['index'], equals(1));
+    expect(callbackResults['isExpanded'], equals(false));
+  });
+
+  testWidgets('didUpdateWidget accounts for toggling between ExpansionPanelList'
+    'and ExpansionPaneList.radio', (WidgetTester tester) async {
+    bool isRadioList = false;
+    final List<bool> _panelExpansionState = <bool>[
+      false,
+      false,
+      false,
+    ];
+
+    await tester.pumpWidget(
+      StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return MaterialApp(
+            home: Scaffold(
+              body: SingleChildScrollView(
+                child: isRadioList ?
```

Comment:
This is a little clunky to read. Maybe `isRadioList ? buildRadioExpansionPanelList : buildExpansionPanelList,` and define two local functions.